### PR TITLE
Return usage, available, staking info for apps

### PR DESCRIPTION
**Pull request by liayoo**
- Extended get state usage APIs to return usage, available, staking info
```
{
   usage: {
     tree_height,
     tree_size,
     tree_bytes
   },
   available: {
     tree_height,
     tree_size,
     tree_bytes
   },
   staking: {
     app,
     total,
     unstakeable
   },
}

```
- Reduced state budgets for the free tier (5% -> 0.5%)

### CODE REVIEW

File: test/unit/db.test.js [hunk 1]
```diff
@@ -3817,7 +3817,7 @@ describe("DB operations", () => {
         const overSizeTx = Transaction.fromTxBody(overSizeTxBody, node.account.private_key);
         const res = node.db.executeTransaction(overSizeTx, false, true, node.bc.lastBlockNumber() + 1);
         assert.deepEqual(res.code, 10902);
-        assert.deepEqual(res.error_message, "Exceeded state budget limit for apps (12621228 > 9000000)");
+        assert.deepEqual(res.error_message.includes("Exceeded state budget limit for apps"),true);
```
Comment (platfowner) [inline test/unit/db.test.js:3820]: 👍

**Comment by platfowner** [comment]
could you consider including in this PR some optimization on getStateFreeTierUsage() using isShallow option?
https://github.com/ainblockchain/ain-blockchain/blob/develop/db/index.js#L1236

File: node/index.js [hunk 2]
```diff
@@ -443,9 +444,62 @@ class BlockchainNode {
     return shardingInfo;
   }
 
-  getStateUsage(appName) {
+  static calcUnstakeableAmount(stateBudget, freeStateBudget, appUsage, freeTierUsage, appStake, totalStake) {
+    if (!totalStake) return appStake;
+    if (!appStake || stateBudget === appUsage) return 0;
```
Comment (platfowner) [inline node/index.js:449]: minor: can '===' be replaced by '<='?

File: node/index.js [hunk 3]
```diff
@@ -443,9 +444,62 @@ class BlockchainNode {
...
+    // NOTE(liayoo): stateUsage <= (appStake - x) / (totalStake - x) * stateBudget
+    const unstakeable = (stateBudget * appStake - appUsage * totalStake) / (stateBudget - appUsage);
```
Comment (platfowner) [inline node/index.js:451]: it looks like the above two lines are not identical??

File: node/index.js (hunk 2 above)
Comment (liayoo) [reply node/index.js]: done! 👍

File: node/index.js (hunk 3 above)
Comment (liayoo) [reply node/index.js:451]: Oh, the x = unstakeable. I'll update the comment!

**Comment by liayoo** [comment]
> could you consider including in this PR some optimization on getStateFreeTierUsage() using isShallow option? https://github.com/ainblockchain/ain-blockchain/blob/develop/db/index.js#L1236

Ah, thanks for the suggestion, but I think isShallow is already there in [L1239](https://github.com/ainblockchain/ain-blockchain/blob/develop/db/index.js#L1239), and [L1241](https://github.com/ainblockchain/ain-blockchain/blob/develop/db/index.js#L1241) only gets a number/null.

File: test/integration/node.test.js [hunk 4]
```diff
@@ -902,7 +917,8 @@ describe('Blockchain Node', () => {
                 "config": {
                   "state": {
                     "max_children": 10,
-                    "gc_max_siblings": 100
+                    "gc_max_siblings": 100,
+                    "gc_num_siblings_deleted": 20,
```
Comment (platfowner) [inline test/integration/node.test.js:921]: oh, thanks for fixing this.👍  i missed this in a hurry.

File: node/index.js (hunk 3 above)
Comment (platfowner) [reply node/index.js:451]: ah, got it. thanks for the update!

**Comment by platfowner** [comment]
> > could you consider including in this PR some optimization on getStateFreeTierUsage() using isShallow option? https://github.com/ainblockchain/ain-blockchain/blob/develop/db/index.js#L1236
> 
> Ah, thanks for the suggestion, but I think isShallow is already there in [L1239](https://github.com/ainblockchain/ain-blockchain/blob/develop/db/index.js#L1239), and [L1241](https://github.com/ainblockchain/ain-blockchain/blob/develop/db/index.js#L1241) only gets a number/null.

you're right, thanks for the links!

**Review by platfowner — APPROVED** [review]
LGTM
Thanks!

**Pull request merged**